### PR TITLE
Dashes should not be allowed in version tag

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-2222

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -653,6 +653,17 @@ def _check_labels(self, repo_info):
...
             pass
 
+        try:
+            _, version_value = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
+            if (version_value and
+                    any(denied in version_value for denied in VERSION_LABEL_FORBIDDEN_CHARS)):
+
+                logger.error("version label '%s' contains not allowed chars : %s", version_value,
+                             VERSION_LABEL_FORBIDDEN_CHARS)
+                raise OsbsValidationException("version label contains not allowed chars")
```
Comment (MartinBasti) [inline osbs/api.py:663]: Can we specify which one?

**Comment by MartinBasti** [comment]
do we know allowed values? I.e. just having check in opposite way?

File: osbs/api.py (hunk 1 above)
Comment (ben-alkov) [reply osbs/api.py:663]: +1 for `denied` instead of `VERSION_LABEL_FORBIDDEN_CHARS`

**Comment by ben-alkov** [comment]
@rcerven; Does this effect users in any way (i.e. do we need a doc update/release note)?

File: osbs/api.py (hunk 1 above)
Comment (rcerven) [reply osbs/api.py:663]: sure

**Comment by rcerven** [comment]
> @rcerven; Does this effect users in any way (i.e. do we need a doc update/release note)?

not really, because that limitation is already in koji, so without this fix, build will fail anyway in koji_import

**Comment by rcerven** [comment]
> do we know allowed values? I.e. just having check in opposite way?

in koji source I've found only raise with '-'

File: osbs/api.py (hunk 1 above)
Comment (MartinBasti) [reply osbs/api.py]: I meant in that exception, so it's shown to a user directly

Comment (rcerven) [reply osbs/api.py]: added there as well :)

File: osbs/api.py [hunk 2]
```diff
@@ -653,6 +653,20 @@ def _check_labels(self, repo_info):
...
+            _, version_value = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
+            if version_value:
+                wrong_chars = \
+                    [denied for denied in VERSION_LABEL_FORBIDDEN_CHARS if denied in version_value]
+
+                if wrong_chars:
+                    logger.error("version label '%s' contains not allowed chars : %s",
+                                 version_value, wrong_chars)
+                    raise OsbsValidationException("version label contains not allowed chars %s" %
+                                                  wrong_chars)
```
Comment (MartinBasti) [inline osbs/api.py:666]:
```suggestion
                    msg = "version label '{}' contains not allowed chars : '{}'".format(version_value, wrong_chars)
                    logger.error(msg)
                    raise OsbsValidationException(msg)
```

File: osbs/api.py [hunk 3]
```diff
@@ -653,6 +653,20 @@ def _check_labels(self, repo_info):
...
+        except KeyError:
+            pass
```
Comment (tkdchen) [inline osbs/api.py:668]: Which line above could raise KeyError? Suggest to write a comment to tell why this error can be ignored.

File: tests/test_api.py [hunk 4]
```diff
@@ -3046,51 +3047,53 @@ def mock_get_image_stream(*args, **kwargs):
...
-                [general]
-                build_json_dir = inputs
-                [default]
-                build_from = image:buildroot:latest
-                openshift_url = /
-                """))
-            fp.flush()
-            config = Configuration(fp.name)
-            osbs_obj = OSBS(config, config)
+    def test_release_and_version_label_validation(self, osbs, label_type, label_value,
```
Comment (tkdchen) [inline tests/test_api.py:3071]: This test case tests invalid release and version label only, but the name `test ... validation` is too generic. Suggest to rename it to something specifically like `test_raise_error_if_release_or_version_label_is_invalid`.

File: osbs/api.py (hunk 2 above)
Comment (rcerven) [reply osbs/api.py:666]: ok

File: osbs/api.py (hunk 3 above)
Comment (rcerven) [reply osbs/api.py:668]: getting version label, added comment

File: tests/test_api.py (hunk 4 above)
Comment (rcerven) [reply tests/test_api.py:3071]: sure

File: osbs/api.py (hunk 3 above)
Comment (MartinBasti) [reply osbs/api.py:669]: I really like:
```python
try:
except:
else:
```
for such cases, when try block is as small as possible

Comment (rcerven) [reply osbs/api.py]: done

**Review by MartinBasti — APPROVED** [review]

**Review by tkdchen — APPROVED** [review]

**Review by ben-alkov — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated: Dashes should not be allowed in version tag

**Pull request merged**